pygmt/tests/test_clib_loading.py

```diff
@@ -72,7 +72,7 @@ def test_load_libgmt_fails(monkeypatch):
     be found.
     """
     with monkeypatch.context() as mpatch:
-        mpatch.setattr(sys, "platform", "win32")  # pretend to be on Windows
+        mpatch.setattr(sys, "platform", "darwin")  # pretend to be on macOS
```

Comment:
Ok, this workaround I applied in a048c8617e2b8e77a79e3743ab76bfbcced0484a doesn't actually work on macOS...

Reply:
> Your workaround [a048c86](https://github.com/GenericMappingTools/pygmt/commit/a048c8617e2b8e77a79e3743ab76bfbcced0484a) fails on macOS CI, because on macOS, PyGMT will find the gmt.dylib successfully in the system default path (I think conda's lib path is in the default search path)
> 
> Two possible workarounds:
> 
>     1. Monkeypatch `sys.platform` so it always returns the wrong platform, e.g., return `linux` on macOS/Windows and `darwin` on Linux
> 
>     2. Instead of monkeypatching `sys.platform`, we can monkeypath `clib_names` to return invalid library names.

Ok, trying out Option 1 in 99fdbf8403bd238226cc752485df1ab1891c2088.